packages/salesforce-adapter/test/change_validators/package_version.test.ts

```
@@ -0,0 +1,192 @@
+        change = toChange({ after: instance })
+      })
+      it('should catch instance', async () => {
+        const expectedErrorMessage = requiresExactVersion
+          ? "Cannot deploy instances with a different package version than target environment's package version"
+          : "Cannot deploy instances with a greater package version than target environment's package version"
+        const errs = await changeValidator([change])
+        expect(errs).toHaveLength(2)
+        expect(errs[0].message).toEqual(expectedErrorMessage)
+        expect(errs[0].elemID).toEqual(getChangeData(change).elemID.createNestedID('packageVersions', '0'))
+        expect(errs[1].message).toEqual(expectedErrorMessage)
+        expect(errs[1].elemID).toEqual(getChangeData(change).elemID.createNestedID('packageVersions', '1'))
+      })
+    },
+  )
+
+  describe.each(Object.entries(TYPES_PACKAGE_VERSION_MATCHING_EXACT_VERSION))(
+    'when package version in the instance is lower than in the target environment',
+    (instanceType: string, exactVersion: ExactVersion) => {
+      let change: Change
+      beforeEach(() => {
+        requiresExactVersion = exactVersion.exactVersion
+        const objectTypeToValidate = createCustomObjectType(instanceType, {
+          elemID: new ElemID('salesforce', instanceType),
+        })
+        const instance = new InstanceElement('asd', objectTypeToValidate, {
+          packageVersions: [
+            {
+              majorNumber: 248,
+              minorNumber: 0,
+              namespace: sbqq,
+            },
+            {
+              majorNumber: 220,
+              minorNumber: 17,
+              namespace: sbaa,
+            },
+            {
+              majorNumber: 1,
+              minorNumber: 7,
+              namespace: fakePackage,
+            },
+          ],
+        })
+        change = toChange({ after: instance })
+      })
+      it('should catch instances that require exact version', async () => {
```

"Catch" is language I would expect around exceptions. Here you can just say "should return an error for all packages that require an exact version match".